### ISSUE TITLE
Panic when interleaving Rust code

### ISSUE THREAD

**Issue by Kobzol** [pacak/cargo-show-asm#65]
Hi! When I run `cargo-show-asm` version `0.1.23`, I get an indexing panic.

Repro:
```bash
$ git clone https://github.com/rust-lang/rustc-perf
$ cd rustc-perf/collector/runtime-benchmarks
$ cargo asm --bin hashmap-bench "hashmap_bench::main" --rust

...
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 1674', /home/kobzol/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-show-asm-0.1.23/src/asm.rs:141:34
stack backtrace:
   0: rust_begin_unwind
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:142:14
   2: core::panicking::panic_bounds_check
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:84:5
   3: cargo_show_asm::asm::dump_range
   4: cargo_show_asm::asm::dump_function
   5: cargo_asm::main
```

Rust version:
```
rustc 1.64.0 (a55dd71d5 2022-09-19)
binary: rustc
commit-hash: a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52
commit-date: 2022-09-19
host: x86_64-unknown-linux-gnu
release: 1.64.0
LLVM version: 14.0.6
```

**Comment by pacak** [comment]
Most likely you don't have rust-src installed, try this:
```
rustup component add rust-src
```
... and I need to figure out what's wrong with the previous fix for this problem :thinking:

**Comment by Kobzol** [comment]
I don't think that's the problem here:
```console
$ rustup component add rust-src
$ info: component 'rust-src' is up to date
```

**Comment by pacak** [comment]
Cool. Can you try running `cargo-asm` with `-vvvv` and see if it prints anything interesting before dying?

**Comment by Kobzol** [comment]
This is the full output:
```console
$ RUST_BACKTRACE=full cargo asm -vvvv --bin hashmap-bench "hashmap_bench::main" --rust
       Fresh version_check v0.9.4
       Fresh unicode-ident v1.0.3
       Fresh cfg-if v1.0.0
       Fresh once_cell v1.13.1
       Fresh autocfg v1.1.0
       Fresh heck v0.4.0
       Fresh os_str_bytes v6.3.0
       Fresh regex-syntax v0.6.27
       Fresh termcolor v1.1.3
       Fresh humantime v2.1.0
       Fresh ryu v1.0.11
       Fresh bitflags v1.3.2
       Fresh textwrap v0.15.0
       Fresh strsim v0.10.0
       Fresh itoa v1.0.3
       Fresh byteorder v1.4.3
       Fresh clap_lex v0.2.4
       Fresh fxhash v0.2.1
       Fresh libc v0.2.132
       Fresh proc-macro2 v1.0.43
       Fresh memchr v2.5.0
       Fresh log v0.4.17
       Fresh anyhow v1.0.62
       Fresh quote v1.0.21
       Fresh getrandom v0.2.7
       Fresh atty v0.2.14
       Fresh aho-corasick v0.7.18
       Fresh perf-event-open-sys v1.0.1
       Fresh syn v1.0.99
       Fresh proc-macro-error-attr v1.0.4
       Fresh ahash v0.7.6
       Fresh regex v1.6.0
       Fresh perf-event v0.4.7
       Fresh hashbrown v0.12.3
       Fresh serde_derive v1.0.143
       Fresh proc-macro-error v1.0.4
       Fresh env_logger v0.9.0
       Fresh indexmap v1.9.1
       Fresh serde v1.0.143
       Fresh clap_derive v3.2.17
       Fresh clap v3.2.17
       Fresh serde_json v1.0.83
       Fresh benchlib v0.1.0 (/tmp/rustc-perf/collector/benchlib)
       Fresh hashmap-bench v0.1.0 (/tmp/rustc-perf/collector/runtime-benchmarks/hashmap)
    Finished release [optimized] target(s) in 0.04s
Scanning "/tmp/rustc-perf/collector/runtime-benchmarks/target/release/deps"
Found some files: ["/tmp/rustc-perf/collector/runtime-benchmarks/target/release/deps/hashmap_bench-6e0a745b74944f0f.s"]
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/ops/function.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/hint.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/rt.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/error.rs"
Reading "/home/kobzol/.cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.62/src/error.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/fmt/mod.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/num/uint_macros.rs"
Reading "/home/kobzol/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/bitmask.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/convert/num.rs"
Reading "/home/kobzol/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/mod.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/ptr/non_null.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/ptr/mut_ptr.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/ptr/mod.rs"
Reading "/home/kobzol/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/map.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/cmp.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/iter/range.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/num/int_macros.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/intrinsics.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/../../stdarch/crates/core_arch/src/x86/sse2.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/alloc.rs"
Reading "/tmp/rustc-perf/collector/runtime-benchmarks/hashmap/src/main.rs"
Reading "/home/kobzol/.cargo/registry/src/github.com-1ecc6299db9ec823/fxhash-0.2.1/lib.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/hash/mod.rs"
Reading "/home/kobzol/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/sse2.rs"
Reading "/home/kobzol/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/alloc.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/raw_vec.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/alloc/layout.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/vec/mod.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/os/fd/owned.rs"
Reading "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/mod.rs"
File not found "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/mod.rs"
Reading "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/bitmask.rs"
File not found "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/bitmask.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/ptr/const_ptr.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/option.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/sync.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/sync/atomic.rs"
Reading "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/sse2.rs"
File not found "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/sse2.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/io/error/repr_bitpacked.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/hash/sip.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/collections/hash/map.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/rc.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/mem/mod.rs"
Reading "/home/kobzol/.cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.62/src/ptr.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/any.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/boxed.rs"
Reading "/tmp/rustc-perf/collector/benchlib/src/benchmark.rs"
Reading "/tmp/rustc-perf/collector/benchlib/src/measure/perf_counter/unix.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/result.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/string.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/slice.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/str.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/iter/adapters/step_by.rs"
Reading "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/map.rs"
File not found "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/map.rs"
Reading "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/alloc.rs"
File not found "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/raw/alloc.rs"
Reading "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/scopeguard.rs"
File not found "/cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/scopeguard.rs"
Reading "/home/kobzol/.cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.3/src/scopeguard.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/slice/cmp.rs"
Reading "/rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/str/traits.rs"
Directive(SectionStart(".text._ZN13hashmap_bench4main17hb40f96645eec7d21E,\"ax\",@progbits"))
.section .text.hashmap_bench::main,"ax",@progbits
Directive(Generic(GenericDirective("p2align\t4, 0x90")))
	.p2align	4, 0x90
Directive(Generic(GenericDirective("type\t_ZN13hashmap_bench4main17hb40f96645eec7d21E,@function")))
	.type	hashmap_bench::main,@function
Label(Label { id: "_ZN13hashmap_bench4main17hb40f96645eec7d21E", local: false })
hashmap_bench::main:
Label(Label { id: ".Lfunc_begin66", local: true })

Directive(Loc(Loc { file: 22, line: 5, column: 0, extra: None }))
		// /tmp/rustc-perf/collector/runtime-benchmarks/hashmap/src/main.rs : 5
		fn main() {
Directive(Generic(GenericDirective("cfi_startproc")))
	.cfi_startproc
Directive(Generic(GenericDirective("cfi_personality 155, DW.ref.rust_eh_personality")))
	.cfi_personality 155, DW.ref.rust_eh_personality
Directive(Generic(GenericDirective("cfi_lsda 27, .Lexception18")))
	.cfi_lsda 27, .Lexception18
Instruction(Instruction { op: "push", args: Some("rbp") })
	push rbp
Directive(Generic(GenericDirective("cfi_def_cfa_offset 16")))
	.cfi_def_cfa_offset 16
Instruction(Instruction { op: "push", args: Some("r15") })
	push r15
Directive(Generic(GenericDirective("cfi_def_cfa_offset 24")))
	.cfi_def_cfa_offset 24
Instruction(Instruction { op: "push", args: Some("r14") })
	push r14
Directive(Generic(GenericDirective("cfi_def_cfa_offset 32")))
	.cfi_def_cfa_offset 32
Instruction(Instruction { op: "push", args: Some("r13") })
	push r13
Directive(Generic(GenericDirective("cfi_def_cfa_offset 40")))
	.cfi_def_cfa_offset 40
Instruction(Instruction { op: "push", args: Some("r12") })
	push r12
Directive(Generic(GenericDirective("cfi_def_cfa_offset 48")))
	.cfi_def_cfa_offset 48
Instruction(Instruction { op: "push", args: Some("rbx") })
	push rbx
Directive(Generic(GenericDirective("cfi_def_cfa_offset 56")))
	.cfi_def_cfa_offset 56
Instruction(Instruction { op: "sub", args: Some("rsp, 360") })
	sub rsp, 360
Directive(Generic(GenericDirective("cfi_def_cfa_offset 416")))
	.cfi_def_cfa_offset 416
Directive(Generic(GenericDirective("cfi_offset rbx, -56")))
	.cfi_offset rbx, -56
Directive(Generic(GenericDirective("cfi_offset r12, -48")))
	.cfi_offset r12, -48
Directive(Generic(GenericDirective("cfi_offset r13, -40")))
	.cfi_offset r13, -40
Directive(Generic(GenericDirective("cfi_offset r14, -32")))
	.cfi_offset r14, -32
Directive(Generic(GenericDirective("cfi_offset r15, -24")))
	.cfi_offset r15, -24
Directive(Generic(GenericDirective("cfi_offset rbp, -16")))
	.cfi_offset rbp, -16
Instruction(Instruction { op: "lea", args: Some("rbx, [rsp + 144]") })
	lea rbx, [rsp + 144]
Label(Label { id: ".Ltmp2263", local: true })

Directive(Loc(Loc { file: 46, line: 10, column: 5, extra: Some("prologue_end") }))
		// /tmp/rustc-perf/collector/benchlib/src/benchmark.rs : 10
		env_logger::Builder::from_default_env()
Instruction(Instruction { op: "mov", args: Some("rdi, rbx") })
	mov rdi, rbx
Instruction(Instruction { op: "call", args: Some("qword ptr [rip + _ZN10env_logger7Builder16from_default_env17hcc9834e33c04fa5dE@GOTPCREL]") })
	call qword ptr [rip + env_logger::Builder::from_default_env@GOTPCREL]
Label(Label { id: ".Ltmp2233", local: true })

Instruction(Instruction { op: "mov", args: Some("esi, 3") })
	mov esi, 3
Instruction(Instruction { op: "mov", args: Some("rdi, rbx") })
	mov rdi, rbx
Instruction(Instruction { op: "call", args: Some("qword ptr [rip + _ZN10env_logger7Builder12filter_level17h69308cabad659ea6E@GOTPCREL]") })
	call qword ptr [rip + env_logger::Builder::filter_level@GOTPCREL]
Label(Label { id: ".Ltmp2234", local: true })

Label(Label { id: ".Ltmp2235", local: true })
Instruction(Instruction { op: "mov", args: Some("rdi, rax") })
	mov rdi, rax
Instruction(Instruction { op: "call", args: Some("qword ptr [rip + _ZN10env_logger7Builder4init17hac08b3457b20a211E@GOTPCREL]") })
	call qword ptr [rip + env_logger::Builder::init@GOTPCREL]
Label(Label { id: ".Ltmp2236", local: true })

Directive(Loc(Loc { file: 46, line: 0, column: 5, extra: Some("is_stmt 0") }))
Instruction(Instruction { op: "lea", args: Some("rdi, [rsp + 144]") })
	lea rdi, [rsp + 144]
Directive(Loc(Loc { file: 46, line: 12, column: 16, extra: Some("is_stmt 1") }))
		// /tmp/rustc-perf/collector/benchlib/src/benchmark.rs : 12
		.init();
Instruction(Instruction { op: "call", args: Some("_ZN4core3ptr40drop_in_place$LT$env_logger..Builder$GT$17h97ef8fdefdddd42eE") })
	call core::ptr::drop_in_place<env_logger::Builder>
Instruction(Instruction { op: "lea", args: Some("rdi, [rsp + 64]") })
	lea rdi, [rsp + 64]
Directive(Loc(Loc { file: 46, line: 13, column: 21, extra: None }))
		// /tmp/rustc-perf/collector/benchlib/src/benchmark.rs : 13
		let mut suite = BenchmarkSuite::new();
Instruction(Instruction { op: "call", args: Some("qword ptr [rip + _ZN8benchlib9benchmark14BenchmarkSuite3new17hdeaaa675730f84c7E@GOTPCREL]") })
	call qword ptr [rip + benchlib::benchmark::BenchmarkSuite::new@GOTPCREL]
Label(Label { id: ".Ltmp2264", local: true })

Directive(Loc(Loc { file: 46, line: 0, column: 21, extra: Some("is_stmt 0") }))
Instruction(Instruction { op: "lea", args: Some("rbx, [rip + .L__unnamed_16]") })
	lea rbx, [rip + .L__unnamed_16]
Instruction(Instruction { op: "mov", args: Some("qword ptr [rsp + 128], rbx") })
	mov qword ptr [rsp + 128], rbx
Instruction(Instruction { op: "mov", args: Some("qword ptr [rsp + 136], 18") })
	mov qword ptr [rsp + 136], 18
Label(Label { id: ".Ltmp2265", local: true })

Directive(Loc(Loc { file: 21, line: 95, column: 14, extra: Some("is_stmt 1") }))
		// /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/alloc.rs : 95
		unsafe { __rust_alloc(layout.size(), layout.align()) }
Instruction(Instruction { op: "mov", args: Some("edi, 16") })
	mov edi, 16
Instruction(Instruction { op: "mov", args: Some("esi, 8") })
	mov esi, 8
Instruction(Instruction { op: "call", args: Some("qword ptr [rip + __rust_alloc@GOTPCREL]") })
	call qword ptr [rip + __rust_alloc@GOTPCREL]
Label(Label { id: ".Ltmp2266", local: true })

Directive(Loc(Loc { file: 21, line: 326, column: 11, extra: None }))
		// /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/alloc.rs : 326
		match Global.allocate(layout) {
Instruction(Instruction { op: "test", args: Some("rax, rax") })
	test rax, rax
Label(Label { id: ".Ltmp2267", local: true })

Directive(Loc(Loc { file: 21, line: 326, column: 5, extra: Some("is_stmt 0") }))
Instruction(Instruction { op: "je", args: Some(".LBB66_3") })
	je .LBB66_3
Label(Label { id: ".Ltmp2268", local: true })

Directive(Loc(Loc { file: 45, line: 215, column: 18, extra: Some("is_stmt 1") }))
		// /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/alloc/src/boxed.rs : 215
		Box::new(x)
Instruction(Instruction { op: "mov", args: Some("qword ptr [rax], rbx") })
	mov qword ptr [rax], rbx
Instruction(Instruction { op: "mov", args: Some("qword ptr [rsp + 40], rax") })
	mov qword ptr [rsp + 40], rax
Instruction(Instruction { op: "mov", args: Some("qword ptr [rax + 8], 18") })
	mov qword ptr [rax + 8], 18
Label(Label { id: ".Ltmp2269", local: true })

Directive(Loc(Loc { file: 45, line: 0, column: 18, extra: Some("is_stmt 0") }))
Instruction(Instruction { op: "mov", args: Some("qword ptr [rsp + 16], rbx") })
	mov qword ptr [rsp + 16], rbx
Instruction(Instruction { op: "mov", args: Some("qword ptr [rsp + 24], 18") })
	mov qword ptr [rsp + 24], 18
Label(Label { id: ".Ltmp2270", local: true })

Directive(Loc(Loc { file: 53, line: 1675, column: 20, extra: Some("is_stmt 1") }))
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 1674', /home/kobzol/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-show-asm-0.1.23/src/asm.rs:141:34
stack backtrace:
   0:     0x55dcf3be1eed - std::backtrace_rs::backtrace::libunwind::trace::h9135f25bc195152c
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x55dcf3be1eed - std::backtrace_rs::backtrace::trace_unsynchronized::h015ee85be510df51
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55dcf3be1eed - std::sys_common::backtrace::_print_fmt::h5fad03caa9652a2c
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x55dcf3be1eed - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2b42ca28d244e5c7
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x55dcf3c0ac7c - core::fmt::write::h401e827d053130ed
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/fmt/mod.rs:1198:17
   5:     0x55dcf3bdce31 - std::io::Write::write_fmt::hffec93268f5cde32
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/io/mod.rs:1672:15
   6:     0x55dcf3be3f05 - std::sys_common::backtrace::_print::h180c4c706ee1d3fb
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x55dcf3be3f05 - std::sys_common::backtrace::print::hd0c35d18765761c9
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x55dcf3be3f05 - std::panicking::default_hook::{{closure}}::h1f023310983bc730
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:295:22
   9:     0x55dcf3be3c21 - std::panicking::default_hook::h188fec3334afd5be
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:314:9
  10:     0x55dcf3be4496 - std::panicking::rust_panic_with_hook::hf26e9d4f97b40096
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:698:17
  11:     0x55dcf3be4387 - std::panicking::begin_panic_handler::{{closure}}::hfab912107608087a
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:588:13
  12:     0x55dcf3be23e4 - std::sys_common::backtrace::__rust_end_short_backtrace::h434b685ce8d9965b
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs:138:18
  13:     0x55dcf3be40b9 - rust_begin_unwind
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:584:5
  14:     0x55dcf354e373 - core::panicking::panic_fmt::ha6dc7f2ab2479463
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:142:14
  15:     0x55dcf354e2b2 - core::panicking::panic_bounds_check::ha6e6615eae13afdc
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:84:5
  16:     0x55dcf35702c5 - cargo_show_asm::asm::dump_range::hec12574c0dcbb4b4
  17:     0x55dcf3571788 - cargo_show_asm::asm::dump_function::ha28bd15ac8b0c1b1
  18:     0x55dcf355e9d0 - cargo_asm::main::hd36060956b78b70a
  19:     0x55dcf35555e3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h583fb4c3e53989a3
  20:     0x55dcf35649c9 - std::rt::lang_start::{{closure}}::h093d5b1184946f62
  21:     0x55dcf3bd643e - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hcdfee62722e5e4b8
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/ops/function.rs:280:13
  22:     0x55dcf3bd643e - std::panicking::try::do_call::h84ca51609826746f
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:492:40
  23:     0x55dcf3bd643e - std::panicking::try::hd58075e533b8e0cb
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:456:19
  24:     0x55dcf3bd643e - std::panic::catch_unwind::h1ebac24d83cb6ce2
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panic.rs:137:14
  25:     0x55dcf3bd643e - std::rt::lang_start_internal::{{closure}}::h0145388a1edd1640
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/rt.rs:128:48
  26:     0x55dcf3bd643e - std::panicking::try::do_call::h7630182e534a0a32
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:492:40
  27:     0x55dcf3bd643e - std::panicking::try::h05b6544f0c6331dc
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:456:19
  28:     0x55dcf3bd643e - std::panic::catch_unwind::h77b2ba8fd3309f34
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panic.rs:137:14
  29:     0x55dcf3bd643e - std::rt::lang_start_internal::h6612c8a7a6861b8b
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/rt.rs:128:20
  30:     0x55dcf355f762 - main
  31:     0x7fbbfded1d90 - __libc_start_call_main
                               at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  32:     0x7fbbfded1e40 - __libc_start_main_impl
                               at ./csu/../csu/libc-start.c:392:3
  33:     0x55dcf354e6c5 - _start
  34:                0x0 - <unknown>
```

**Comment by pacak** [comment]
I see, looks like it can't access some of the `bashbrown` files, not sure what makes them special. But I can reproduce the problem locally so will fix

**Comment by pacak** [comment]
This works for me, can you check?
`cargo install --git https://github.com/pacak/cargo-show-asm --branch custom`

**Comment by Kobzol** [comment]
Hmm, cargo install has some problem with selecting the package to install:
```console
$ cargo install --git https://github.com/pacak/cargo-show-asm --branch custom
    Updating git repository `https://github.com/pacak/cargo-show-asm`
error: multiple packages with binaries found: cargo-show-asm, sample. When installing a git repository, cargo will always search the entire repo for any Cargo.toml. Please specify which to install.
$ cargo install --git https://github.com/pacak/cargo-show-asm --branch custom --bin cargo-show-asm
(same output)
$ cargo install --git https://github.com/pacak/cargo-show-asm --branch custom --bin cargo-asm
(same output)
$ cargo install --git https://github.com/pacak/cargo-show-asm --branch custom --bins
(same output)
```

Anyway, I compiled the branch manually and tested it. It works! :tada: Thank you very much.

I'm planning to use `cargo-show-asm` for diffing the codegen of runtime benchmarks that we're currently adding to the `rustc` perf suite. Thank you very much for this tool!

**Comment by pacak** [comment]
> Hmm, cargo install has some problem with selecting the package to install:

Hmm.... Right, my bad, there's also a crate I'm using in CI. Probably need to change it to be a library to make the installation less confusing.

**Comment by pacak** [comment]
> I'm planning to use `cargo-show-asm` for diffing the codegen of runtime benchmarks that we're currently adding to the rustc perf suite.

Any feedback on diffing specifically would be interesting. I had this https://github.com/pacak/cargo-show-asm/issues/31 but maybe something else can be improved as well.

**Comment by Kobzol** [comment]
Ok, it will take me some time, but I'll let you know. What set of flags would you recommend for diffable output? `--keep-labels`?

**Comment by pacak** [comment]
Default should be more diffable, `--keep-labels` is something to see the original code with least modifications. But if you notice anything that something unnecessary changing and have a rough idea how to remove that - this would be useful.

**Comment by pacak** [comment]
Fixed and published 0.1.24, thank you for the bugreport.

**Comment by Kobzol** [comment]
Thank you for the quick fix!
